src/main/java/uk/gov/hmcts/reform/cwrdapi/config/FeignInterceptorConfiguration.java

```diff
@@ -0,0 +1,40 @@
+package uk.gov.hmcts.reform.cwrdapi.config;
+
+import feign.RequestInterceptor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.web.context.request.RequestContextHolder;
+import org.springframework.web.context.request.ServletRequestAttributes;
+
+import java.util.Enumeration;
+import javax.servlet.http.HttpServletRequest;
+
+@Slf4j
+public class FeignInterceptorConfiguration {
+
+    @Value("${loggingComponentName}")
+    private String loggingComponentName;
+
+    @Bean
+    public RequestInterceptor requestInterceptor(FeignHeaderConfig config) {
+        return requestTemplate -> {
+            ServletRequestAttributes attrs = (ServletRequestAttributes) RequestContextHolder.getRequestAttributes();
+            if (attrs != null) {
+                HttpServletRequest request = attrs.getRequest();
+                Enumeration<String> headerNames = request.getHeaderNames();
+                if (headerNames != null) {
+                    while (headerNames.hasMoreElements()) {
+                        String name = headerNames.nextElement();
```

Comment:
We can refactor it better with Java lamda. please avoid if else and use lamda filter.

Reply:
i just used from prd and it involved lot conditions and later will try to implement  if possible